## Sentiment Analysis using TextBlob

This library is very simple and easy to use for obtaining a quick sentiment analysis, however it is not always very accurate. More on that later. 

In [2]:
import csv
from textblob import TextBlob
import pandas as pd
import numpy as np
pd.set_option('display.max_rows', 500)
df = pd.read_csv("airbnb_reviews.csv")

The code is all based on the quickstart guide from the official [documentation](https://textblob.readthedocs.io/en/dev/quickstart.html) of the library.

In [ ]:
Polarity = []
Sentiment = []
# Categorizing sentiment
for row in df['comments']:
    feeds = TextBlob(row)
    polarity = feeds.sentiment.polarity
    if(feeds.sentiment.polarity > 0.5):
        sentiment = "Positve"
    elif(feeds.sentiment.polarity < 0.5):
        sentiment = "Negative"
    else:
        sentiment = "Neutral"
# Add data to list
    Polarity.append(polarity)
    Sentiment.append(sentiment)
# Append data to DF
df['Polarity'] = Polarity
df['Sentiment'] = Sentiment
# Export to csv
df.to_csv('airbnb_reviews_sentiment.csv',index=False, encoding = 'utf-8')

### Why I say it is not very accurate ?
TextBlob will score a word from -1 to 1, being -1 the most negative and +1 the most positive.

In [3]:
print(TextBlob("Great location, gorgeous apartment!").sentiment)

Sentiment(polarity=0.8375, subjectivity=0.825)


For many reviews, mostly the ones that had no ambiguity in its words, the score can be quite accurate however that's not always the case.

In [2]:
print(TextBlob("You will not be disappointed, go ahead and book!").sentiment)

Sentiment(polarity=-0.9375, subjectivity=0.75)


It is a good a review, however TextBlob polarity is **-0.9375**. That is due to the way this library works which basically uses a dictionary of words, amplifiers, modifiers, negations among other fancy terms that added together give a score to the whole sentence and finally a polarity.

More details about this can be found in this [post](https://planspace.org/20150607-textblob_sentiment/). from planspace.org. 

Despite this I will continue with the next part and use this data anyways. (In a newer version I might improve this.. maybe? )